src/main/java/com/commercetools/sync/products/ProductSync.java

```diff
@@ -332,16 +334,21 @@ public ProductSync(@Nonnull final ProductSyncOptions productSyncOptions) {
                     .orElseGet(() -> applyCallbackAndCreate(resolvedDraft));
 
             })
-
             .exceptionally(completionException -> {
-
-                final ReferenceResolutionException referenceResolutionException =
-                    (ReferenceResolutionException) completionException.getCause();
-                final String errorMessage = format(FAILED_TO_RESOLVE_REFERENCES, newProductDraft.getKey(),
-                    referenceResolutionException.getMessage());
-                handleError(errorMessage, referenceResolutionException, 1);
-
-                return null;
+                if (completionException.getCause() instanceof ReferenceResolutionException) {
```

Comment:
the part related to the exception handling seems duplicated with the CartDiscountSync logic except for different keys and different messages, maybe avoid the duplication so when we needed to change that logic again we would not need to refactor all Sync instances.

Reply:
As discussed with Roman, the thrown exception will be passed into handleError directly and let user to deal with it. If-else checking is no longer used in Sync classes.